src/Delete.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Message\User;
+
+use Message\Cog\DB;
+use Message\Cog\ValueObject\DateTimeImmutable;
+
+/**
+ * Decorator class for deleting users.
+ *
+ * @author Grace Cooper <grace@message.co.uk>
+ */
+class Delete
+{
+	private $_transaction;
+	private $_transOverride = false;
+	private $_currentUser;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param DB\Transaction $transaction
+	 * @param DB\Query             $query           The database query instance to use
+	 * @param User                $user            The currently logged in user
+	 */
+	public function __construct(
+		DB\Transaction $transaction,
+		UserInterface $user)
+	{
+		$this->_transaction     = $transaction;
+		$this->_currentUser     = $user;
+	}
+
+	public function setTransaction(DB\Transaction $transaction)
+	{
+		$this->_transaction = $transaction;
+		$this->_transOverride = true;
+	}
+
+	public function delete(User $user)
+	{
+		$this->_delete($user);
+		$this->_commit();
+	}
+
+	/**
+	 * Set a user to deleted
+	 *
+	 * @param  User   $user        The user to change the details for
+	 *
+	 * @author Grace Cooper
+	 */
+	private function _delete(User $user)
+	{
+		$user->authorship->delete(null, $this->_currentUser->id);
```

Comment:
The Authorship object is annoying, but the second parameter should be the user, not the user ID